src/ngraph_builder.cc

```diff
@@ -314,6 +314,136 @@ tf::Status Builder::TranslateGraph(const std::vector<tf::TensorShape>& inputs,
       ng_op_map[op->name()] = ng_avgpool;
     }
     // -------
+    // BatchMatMul
+    // -------
+    else if (op->type_string() == "BatchMatMul") {
+      if (op->num_inputs() != 2) {
+        return tf::errors::InvalidArgument(
+            "Number of inputs is not 2 for BatchMatMul");
+      }
+ 
+      tf::Node* tf_lhs;
+      tf::Node* tf_rhs;
+      TF_RETURN_IF_ERROR(op->input_node(0, &tf_lhs));
+      TF_RETURN_IF_ERROR(op->input_node(1, &tf_rhs));
+  
+      try {
+        ng_op_map.at(tf_lhs->name());
+      } catch(const std::out_of_range&) {
+          return tf::errors::NotFound(tf_lhs->name(),"is not found in ng_op_map!");
+      }
+
+      try {
+        ng_op_map.at(tf_rhs->name());
+      } catch(const std::out_of_range&) {
+          return tf::errors::NotFound(tf_rhs->name(),"is not found in ng_op_map!");
+      }
+
+      auto ng_lhs = ng_op_map.at(tf_lhs->name()); 
+      auto ng_rhs = ng_op_map.at(tf_rhs->name()); 
+      auto ng_lhs_shape = ng_lhs->get_shape(); 
+      auto ng_rhs_shape = ng_rhs->get_shape();
+
+      if(ng_lhs_shape.size() != ng_rhs_shape.size()) {
+        return tf::errors::InvalidArgument(
+            "Dimensions of two input args are not the same for BatchMatMul");
+      }
+      size_t n_dims = ng_lhs_shape.size();
+      if(n_dims < 2) {
+        return tf::errors::InvalidArgument(
+            "Dimensions of input args for BatchMatMul must be >=2", n_dims);
+      }
+
+      ng::AxisVector out_axes;
+      for (size_t i = 0; i < n_dims - 2; ++i) {
+        if(ng_lhs_shape[i] != ng_rhs_shape[i]){
+          return tf::errors::InvalidArgument(
+              "ng_lhs_shape and ng_rhs_shape must be the same for BatchMatMul for each dimension",
+               i);
+        }
+        out_axes.push_back(i);
+      }
+
+      bool tf_adj_x = false;
+      bool tf_adj_y = false;
+      TF_RETURN_IF_ERROR(tf::GetNodeAttr(op->attrs(), "adj_x", &tf_adj_x));
```

Comment:
Let's add some unit tests for cases where `adj_x` and/or `adj_y` are true.

Reply:
Added the unit test for adj_x and adj_y